### PR TITLE
Download logs

### PR DESCRIPTION
**Pull request by ThetaSinner**

### CODE REVIEW

File: crates/trycp_api/src/lib.rs [hunk 1]
```diff
@@ -153,3 +159,22 @@ pub enum MessageToClient {
         response: std::result::Result<MessageResponse, String>,
     },
 }
+
+/// Messages returned directly by the TryCp server, rather than relayed from Holochain
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "snake_case", tag = "type", content = "data")]
+pub enum TryCpServerResponse {
```
Comment (ThetaSinner) [inline crates/trycp_api/src/lib.rs:166]: Some type f*ry required to let the JS client decode these types. It's the first time we've returned anything more complicated than a string and the JS client has a special case for strings.

File: crates/trycp_server/src/download_logs.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    let lair_stderr = player_dir.join(LAIR_STDERR_LOG_FILENAME);
+    let lair_stderr = std::fs::read(&lair_stderr).context(LairStdErr { id: id.clone() })?;
+
+    let conductor_stdout = player_dir.join(CONDUCTOR_STDOUT_LOG_FILENAME);
+    let conductor_stdout =
+        std::fs::read(&conductor_stdout).context(HolochainStdout { id: id.clone() })?;
+
+    let conductor_stderr = player_dir.join(CONDUCTOR_STDERR_LOG_FILENAME);
+    let conductor_stderr = std::fs::read(&conductor_stderr).context(HolochainStderr { id })?;
```
Comment (neonphog) [inline crates/trycp_server/src/download_logs.rs:45]: are we cool with these blocking whatever thread happens to be running?

**Review by neonphog — APPROVED** [review]
lgtm

File: crates/trycp_server/src/download_logs.rs (hunk 2 above)
Comment (ThetaSinner) [reply crates/trycp_server/src/download_logs.rs:45]: The entire function is being run on a blocking thread -> https://github.com/holochain/tryorama/pull/222/files#diff-db8eb8bbb5e02b33e796b281bce321e0d781f68204bac8f217e5eddbe9e34319R256

This seems to be the established pattern for the trycp_server, to offload the entire function if it's not async

**Pull request merged**